### PR TITLE
add process guidelines to contributing docs

### PR DESCRIPTION
**Pull request by ashygee**
Guidelines to communicate our new octicons design process has been added to our CONTRIBUTING.md docs.

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -2,6 +2,37 @@
 
 Thank you for your interest in contributing to Octicons! We are currently only accepting submissions from GitHub staff and only include icons that are used in the GitHub UI.
 
+## Icon request process
+
+### 1. Icon request is made
+
+- Icon requests are made using the issue template in the primer/octicons repo. If the icon being requested is for a product/feature that has not yet been release, please use the [template](https://github.com/github/design-systems/issues/new?assignees=ashygee%2C+edokoa&labels=area%3A+icons%2C+type%3A+request&template=octicon_request.md&title=%5BIcon+request%5D+) found in the [github/design-systems repo](https://github.com/github/design-systems/issues/new/choose).
```
Comment (colebemis) [inline .github/CONTRIBUTING.md:9]: It'd be nice to mention that we don't accept icon request from outside of GitHub here.

**Review by colebemis — APPROVED** [review]
Looks good! Just left one comment

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (ashygee) [reply .github/CONTRIBUTING.md:9]: @colebemis that's mentioned here https://github.com/primer/octicons/pull/547/files#diff-98e64bc1cd2db9333c6effe87bbe0d6dfe8714aba4c6bde45aa037fe0796e44cR3

Comment (colebemis) [reply .github/CONTRIBUTING.md:9]: Oh duh 🤦

**Pull request merged**